diff --git a/kernel/nvidia/nv-dmabuf.c b/kernel/nvidia/nv-dmabuf.c
index f060839..2146f53 100644
--- a/kernel/nvidia/nv-dmabuf.c
+++ b/kernel/nvidia/nv-dmabuf.c
@@ -23,6 +23,10 @@
 #include <linux/dma-buf.h>
 #include "nv-dmabuf.h"
 
+#if defined(MODULE_IMPORT_NS)
+MODULE_IMPORT_NS(DMA_BUF);
+#endif
+
 #if defined(CONFIG_DMA_SHARED_BUFFER)
 
 typedef struct nv_dma_buf_mem_handle
diff --git a/kernel/nvidia/os-interface.c b/kernel/nvidia/os-interface.c
index 4168b03..d4bb99b 100644
--- a/kernel/nvidia/os-interface.c
+++ b/kernel/nvidia/os-interface.c
@@ -45,7 +45,11 @@ NvU8  os_page_shift = PAGE_SHIFT;
 NvU32 os_sev_status = 0;
 NvBool os_sev_enabled = 0;
 
+#if defined(CONFIG_DMA_SHARED_BUFFER)
+NvBool os_dma_buf_enabled = NV_TRUE;
+#else
 NvBool os_dma_buf_enabled = NV_FALSE;
+#endif // CONFIG_DMA_SHARED_BUFFER
 
 void NV_API_CALL os_disable_console_access(void)
 {
