diff --git a/kernel/nvidia-drm/nvidia-drm-drv.c b/kernel/nvidia-drm/nvidia-drm-drv.c
index e48378f..5b5ed2b 100644
--- a/kernel/nvidia-drm/nvidia-drm-drv.c
+++ b/kernel/nvidia-drm/nvidia-drm-drv.c
@@ -296,6 +296,10 @@ static void nv_drm_enumerate_encoders_and_connectors
             "Failed to enumurate NvKmsKapiDisplay count");
     }
 
+    NV_DRM_DEV_LOG_INFO(
+            nv_dev,
+            "nv_drm_enumerate_encoders_and_connectors nDisplays=%u", nDisplays);
+
     if (nDisplays != 0) {
         NvKmsKapiDisplay *hDisplays =
             nv_drm_calloc(nDisplays, sizeof(*hDisplays));

